fast64_internal/oot/oot_utility.py

```diff
@@ -599,3 +630,260 @@ def oot_utility_register():
 def oot_utility_unregister():
 	for cls in reversed(oot_utility_classes):
 		unregister_class(cls)
+
+def getActorParameter(object, field, shift, mask):
+	'''Returns the actor parameter with the shifts'''
+	attr = getattr(object, field, '0x00')
+	if attr != '0x00' and attr is not None:					
+		if isinstance(attr, str):
+			if shift != '0':
+				return f"(({attr} << {shift}) & {mask}) | "
+			else:
+				return f"({attr} & {mask}) | " 
+		elif isinstance(attr, bool) and attr:
+			return f"(1 << {shift}) | "
+		else:
+			return ""
```

Comment:
is there a case in which attr being neither str nor bool is not an error?
maybe this should raise PluginError / NotImplementedError instead of defaulting to ""

Reply:
I don't really remember what I intended there, I think I thought it was useless to raise an error, I can't tell why though

I can't get a proper opinion on this right now but you're probably right

Reply:
I did figured out why I did it that way and fixed it